db/models.py

```diff
@@ -46,3 +53,71 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class User(AbstractUser):
+    pass
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        "MovieSession",
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    order = models.ForeignKey(
+        "Order",
+        on_delete=models.CASCADE,
+        related_name="tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=("row", "seat", "movie_session"),
+                                    name="unique_ticket_rows_seats_sessions")
+        ]
+
+    def __str__(self) -> str:
+        return f"{self.movie_session} (row: {self.row}, seat: {self.seat})"
+
+    def clean(self) -> None:
+        if not (self.row <= self.movie_session.cinema_hall.rows):
+            raise ValidationError({"row": [
+                f"row number must be in available range: (1, rows): "
+                f"(1, {self.movie_session.cinema_hall.rows})"]})
+        if not (self.seat <= self.movie_session.cinema_hall.seats_in_row):
+            raise ValidationError({"seat": [
+                f"seat number must be in available range: (1, seats_in_row): "
+                f"(1, {self.movie_session.cinema_hall.seats_in_row})"
+            ]})
+
+    def save(
+            self,
+            force_insert: bool = False,
+            force_update: bool = False,
+            using: str = None,
+            update_fields: list = None
+    ) -> None:
+        self.full_clean()
+        return super(Ticket, self).save(
+            force_insert=False,
+            force_update=False,
+            using=None,
+            update_fields=None
+        )
```

Comment:
django docs use the args and kwargs for `save()` and without any argument for `super()`, so we should do the same.

See example [here](https://docs.djangoproject.com/en/4.2/ref/models/instances/#customizing-model-loading).

Reply:
Thanks!